### PR TITLE
BGP CP: API Helper Functions Cleanup

### PR DESCRIPTION
**Pull request by danehans**
- Moves functions next to types.
- Renames functions to ParseXXX
- Adds godocs.

### CODE REVIEW

**Review by asauber — APPROVED** [review]

**Comment by rastislavs** [comment]
Thanks! in general I like this cleanup (especially considering it together with https://github.com/cilium/cilium/pull/28035).
Just one idea. As we are expanding `pkg/bgpv1/types/state.go` here, I think the name of this file no longer matches with what it is. That might also be the reason why some of the conversion funcs were duplicated. Not sure what should be a better name though ATM :)

**Comment by danehans** [comment]
@rastislavs thanks for the review and feedback. I documented my ideas for reorganizing the gobgp pkg in [this](https://docs.google.com/document/d/1VLuvhaz3K879efymmqiCGzGMCedeD0KqdLCS5wT3Dkc/edit?usp=sharing) Google doc. PTAL when you have a moment and provide feedback. I'm happy to use this PR to implement the larger reorganization based on the outcome of the Google doc.

**Comment by rastislavs** [comment]
> @rastislavs thanks for the review and feedback. I documented my ideas for reorganizing the gobgp pkg in [this](https://docs.google.com/document/d/1VLuvhaz3K879efymmqiCGzGMCedeD0KqdLCS5wT3Dkc/edit?usp=sharing) Google doc. PTAL when you have a moment and provide feedback. I'm happy to use this PR to implement the larger reorganization based on the outcome of the Google doc.

The changes in the doc are reasonable, but as those are constrained in gobgp package, I think it makes sense to open a PR for them separately.

File: pkg/bgpv1/api/conversions.go [hunk 1]
```diff
@@ -16,85 +16,76 @@ import (
...
-		return types.AfiIPv4
-	case "ipv6":
-		return types.AfiIPv6
-	case "l2vpn":
-		return types.AfiL2VPN
-	case "ls":
-		return types.AfiLS
-	case "opaque":
-		return types.AfiOpaque
+func ToAPIAfi(afi types.Afi) string {
```
Comment (rastislavs) [inline pkg/bgpv1/api/conversions.go]: Why this adds back the code removed in https://github.com/cilium/cilium/pull/28035?

Comment (danehans) [reply pkg/bgpv1/api/conversions.go]: Thanks for catching this. It must have slipped back in during a rebase.

**Review by rastislavs — APPROVED** [review]
LGTM

**Pull request merged**